data/console/check_registration_status.py

```diff
@@ -28,27 +28,38 @@
 # or "Registered" products.
 # Return the account of missing matched product
 ret = 0
-for prod in json.loads(stdout, encoding="utf-8"):
-    prod['base_version'] = prod['version'].split('.')[0]
-    prod['result'] = 'match'
-
-    # Module version is not service pack specific
-    if prod['identifier'].find('module') != -1:
-        if dist['BASE_VERSION'] != prod['base_version']:
-            prod['result'] = 'mismatch'
-    else:
-        if dist['VERSION_ID'] != prod['version']:
-            # Live Patching is seen as Module before SLE12SP3, then seen as extension since SLE12SP3
-            # So its version should equal the base product's base version before SLE12SP3, while equal 
-            # the base product's version since SLE12SP3. 
-            if dist['VERSION_ID'] >= '12.3':
+try:
+    for prod in json.loads(stdout, encoding="utf-8"):
+        prod['base_version'] = prod['version'].split('.')[0]
+        prod['result'] = 'match'
+
+        # Module version is not service pack specific
+        if prod['identifier'].find('module') != -1:
+            if dist['BASE_VERSION'] != prod['base_version']:
                 prod['result'] = 'mismatch'
-            else:
-                if dist['BASE_VERSION'] != prod['version']:
-                   prod['result'] = 'mismatch'
+        else:
+            if dist['VERSION_ID'] != prod['version']:
+                # Live Patching is seen as Module before SLE12SP3, then seen as extension since SLE12SP3
+                # So its version should equal the base product's base version before SLE12SP3, while equal 
+                # the base product's version since SLE12SP3. 
+                if dist['VERSION_ID'] >= '12.3':
+                    prod['result'] = 'mismatch'
+                else:
+                    if dist['BASE_VERSION'] != prod['version']:
+                        prod['result'] = 'mismatch'
 
-    if prod['result'] == 'mismatch':
-        ret += 1
-    print("Product %(identifier)s with version %(version)s is %(status)s [%(result)s]" % prod)
+        if prod['result'] == 'mismatch':
+            ret += 1
+        print("Product %(identifier)s with version %(version)s is %(status)s [%(result)s]" % prod)
+except ValueError as err:
+    # Workaround of poo#42845
+    # In online migration tests using proxy SCC, repos' urls will be invalid
+    # after snapper rollback to the snapshot before migration.  In such case,
+    # stdout will not be in json but in xml format, causing ValueError in
+    # line 32.
+    # Lines below catches this exact error, preventing the script from crashing
+    invalid_url_warning = 'Valid metadata not found at specified URL'
+    if invalid_url_warning in stdout.decode('utf-8'):
+        print('Script exiting: %s\nInvalid repo URL, trigger proxy scc workaround' % err)
```

Comment:
So this causes the test to die? or is the workaround handled somewhere else?

Reply:
Hi, it's this line (https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/71d40b36d6072aaf9c44b48c79686dccfe4bda03/data/console/check_registration_status.py#L31)

It assumes that variable `stdout` must be in JSON format. When `stdout` is in XML, a ValueError will be rasied, causing the script to exit and making the test to die.

My code uses `try … except` to catch this error and let the test pass.